RecoVertex/V0Producer/src/V0Fitter.cc

```diff
@@ -240,13 +243,17 @@ void V0Fitter::fitAll(const edm::Event& iEvent, const edm::EventSetup& iSetup,
       GlobalVector negativeP(trajMins->momentum());
       GlobalVector totalP(positiveP + negativeP);
 
-      // calculate the pointing angle
-      double posx = theVtx.x() - beamSpotPos.x();
-      double posy = theVtx.y() - beamSpotPos.y();
-      double momx = totalP.x();
-      double momy = totalP.y();
-      double pointangle = (posx*momx+posy*momy)/(sqrt(posx*posx+posy*posy)*sqrt(momx*momx+momy*momy));
-      if (pointangle < v0CosThetaCut_) continue;
+      // 2D pointing angle
+      TVector3 pointXY(theVtx.x()-referencePos.x(), theVtx.y()-referencePos.y(), 0.);
+      TVector3 momXY(totalP.x(), totalP.y(), 0.);
+      double angleXY = pointXY.Angle(momXY);
+      if (cos(angleXY) < cosThetaXYCut_) continue;
```

Comment:
i believe that may be responsible for loss of one kshort out of a few
thousands and gaining one lambda out of like 700

On Tue, Oct 6, 2015 at 2:52 PM, Slava Krutelyov <notifications@github.com>
wrote:

> In RecoVertex/V0Producer/src/V0Fitter.cc
> <https://github.com/cms-sw/cmssw/pull/11660#discussion_r41306178>:
>
> > @@ -240,13 +243,17 @@ void V0Fitter::fitAll(const edm::Event& iEvent, const edm::EventSetup& iSetup,
> >        GlobalVector negativeP(trajMins->momentum());
> >        GlobalVector totalP(positiveP + negativeP);
> >
> > -      // calculate the pointing angle
> > -      double posx = theVtx.x() - beamSpotPos.x();
> > -      double posy = theVtx.y() - beamSpotPos.y();
> > -      double momx = totalP.x();
> > -      double momy = totalP.y();
> > -      double pointangle = (posx*momx+posy*momy)/(sqrt(posx*posx+posy*posy)*sqrt(momx*momx+momy*momy));
> > -      if (pointangle < v0CosThetaCut_) continue;
> > +      // 2D pointing angle
> > +      TVector3 pointXY(theVtx.x()-referencePos.x(), theVtx.y()-referencePos.y(), 0.);
> > +      TVector3 momXY(totalP.x(), totalP.y(), 0.);
> > +      double angleXY = pointXY.Angle(momXY);
> > +      if (cos(angleXY) < cosThetaXYCut_) continue;
>
> dot-product with normalization would have worked as well and also come out
> faster and more precise; the cost is probably not very significant here; no
> strong preference to change it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/11660/files#r41306178>.
>